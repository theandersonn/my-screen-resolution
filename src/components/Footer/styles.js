import styled, { css } from 'styled-components'

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    margin-top: auto;
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.white};

    > svg {
      height: 20px;
      width: 20px;
      vertical-align: middle;
      color: red;
    }

    > a {
      text-decoration: none;
      color: ${theme.colors.white};

      &:hover {
        color: ${theme.colors.secondary};
      }
    }
  `}
`
