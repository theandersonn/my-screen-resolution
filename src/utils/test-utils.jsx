import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

const customRender = (children) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export * from '@testing-library/react'
export { customRender as render }
