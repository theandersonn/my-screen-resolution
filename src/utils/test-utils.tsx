import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'styles/theme';

const customRender = (ui: React.ReactElement, options = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);

export * from '@testing-library/react';
export { customRender as render };
