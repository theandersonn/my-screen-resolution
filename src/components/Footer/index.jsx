import * as S from './styles'

const Footer = () => {
  return <S.Wrapper>Made with [heart] by theandersonn</S.Wrapper>
}

export default Footer
