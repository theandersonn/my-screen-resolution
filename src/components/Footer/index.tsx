import { TiHeart as HeartIcon } from 'react-icons/ti';

import * as S from './styles';

const Footer = () => {
  return (
    <S.Wrapper>
      Made with <HeartIcon /> by {` `}
      <a
        href="https://github.com/theandersonn"
        target="_blank"
        rel="noreferrer"
      >
        theandersonn
      </a>
    </S.Wrapper>
  );
};

export default Footer;
