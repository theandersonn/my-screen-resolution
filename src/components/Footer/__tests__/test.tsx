import { screen } from '@testing-library/react';
import { render } from 'utils/test-utils';

import Footer from '..';

describe('<Footer />', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />);

    expect(screen.getByRole('link', { name: /theandersonn/i })).toHaveAttribute(
      'href',
      'https://github.com/theandersonn'
    );
  });
});
