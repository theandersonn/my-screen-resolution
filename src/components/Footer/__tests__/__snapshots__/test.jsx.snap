// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer
  class="sc-bdvvaa gMlHxz"
>
  Made with 
  <svg
    baseProfile="tiny"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    version="1.2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 10.375c0-2.416-1.959-4.375-4.375-4.375s-4.375 1.959-4.375 4.375c0 1.127.159 2.784 1.75 4.375l7 5.25s5.409-3.659 7-5.25 1.75-3.248 1.75-4.375c0-2.416-1.959-4.375-4.375-4.375s-4.375 1.959-4.375 4.375"
    />
  </svg>
   by 
   
  <a
    href="https://github.com/theandersonn"
    rel="noreferrer"
    target="_blank"
  >
    theandersonn
  </a>
</footer>
`;
